### DISCUSSION TITLE
Why show oldest version on the master branch?

### DISCUSSION THREAD

**Discussion by nmgwddj** [GitTools/GitVersion#2846] [Q&A] [answered]
`git branch -a` result
```
  develop
* master
  remotes/origin/HEAD -> origin/master
  remotes/origin/develop
  remotes/origin/master
```

`git tag` result:
```
8.3.0
8.3.1
8.3.5
8.4.0
8.4.1
8.5.0
8.5.1
8.6.2
8.7.0
8.7.1
```

When I run `gitversion` on the master branch, it returns the oldest tag as the version number
```
{
  "Major": 8,
  "Minor": 4,
  "Patch": 0,
  "PreReleaseTag": "",
  "PreReleaseTagWithDash": "",
  "PreReleaseLabel": "",
  "PreReleaseLabelWithDash": "",
  "PreReleaseNumber": null,
  "WeightedPreReleaseNumber": 60000,
  "BuildMetaData": null,
  "BuildMetaDataPadded": "",
  "FullBuildMetaData": "Branch.master.Sha.0be42b9ae35dc9df10a018bca25318ae68edf443",
  "MajorMinorPatch": "8.4.0",
  "SemVer": "8.4.0",
  "LegacySemVer": "8.4.0",
  "LegacySemVerPadded": "8.4.0",
  "AssemblySemVer": "8.4.0.0",
  "AssemblySemFileVer": "8.4.0.0",
  "FullSemVer": "8.4.0",
  "InformationalVersion": "8.4.0+Branch.master.Sha.0be42b9ae35dc9df10a018bca25318ae68edf443",
  "BranchName": "master",
  "EscapedBranchName": "master",
  "Sha": "0be42b9ae35dc9df10a018bca25318ae68edf443",
  "ShortSha": "0be42b9",
  "NuGetVersionV2": "8.4.0",
  "NuGetVersion": "8.4.0",
  "NuGetPreReleaseTagV2": "",
  "NuGetPreReleaseTag": "",
  "VersionSourceSha": "0be42b9ae35dc9df10a018bca25318ae68edf443",
  "CommitsSinceVersionSource": 89,
  "CommitsSinceVersionSourcePadded": "0089",
  "UncommittedChanges": 223,
  "CommitDate": "2021-04-26"
}
```

**Comment by asbjornu** [comment]
GitVersion should order version numbers by value, not by date. From your list of tags I would therefore expect it to chose `8.7.1`. Are all tags made on commits in the `master` branch? Are any commits being ignored in `GitVersion.yml`?

**Comment by nmgwddj** [comment]
This is an old project. GitVersion has never been used for management before. We hope to integrate it into CI and use GitVersion to manage product versions and CI package names.
In the past, gitflow was used by the client to execute `git flow release finish` and then push each branch and tag to the remote end.

**Comment by asbjornu** [comment, marked as answer]
With `git flow` I think the tags suddenly were switched from being made on `master` / `main` to the `release/*` branch for some inexplicable reason. That means the tags aren't on the `master` branch and thus not accessible to GitVersion as it currently operates.

Please see #2206 and nvie/gitflow#432 for details.
